modules/reference/pages/console/record-deserialization.adoc

```diff
@@ -1,27 +1,171 @@
+into a binary wrapper that contains meta information like the schema ID or the used prototypes. Therefore, the application
+that deserializes the Kafka records must recognize the format. Redpanda Console supports both formats. The deserialization process
+requires Redpanda Console to be aware of the used `.proto` files as well as a mapping of what prototype should be used for each topic.
+This information can either be sourced from the Schema Registry or it can be provided with additional configuration so the files
+can be pulled from the local file system or a GitHub repository.
+
+NOTE: To support imports, all prototypes are first registered in a proto registry.
+You must ensure that all imported prototypes are part of the repository. Standard types (such as Google's timestamp type) are included by default and don't need to be added.
+
+=== Schema Registry
+
+Messages that have been serialized using Confluent's KafkaProtobufSerializer can only be deserialized if the Schema Registry is configured.
+Unlike other providers, the Schema Registry does not require you to set up mappings that define which topics use which prototypes. Instead,
+this information is inferred from the messages, and the Schema Registry finds the right prototype for deserialization.
+
+To set up the Schema Registry in Redpanda Console with self-hosted Redpanda, see xref:manage:schema-reg/schema-reg-ui.adoc[]. The Protobuf deserializer uses the same Schema Registry client that is configured under `kafka.schemaRegistry`. 
+A valid configuration looks like the following:
+
+[,yaml]
+----
+kafka:
+  schemaRegistry:
+    enabled: true
+    urls: ["https://my-schema-registry.com"]
+    username: console
+    password: redacted # Or set using flags or env variable
+  protobuf:
+    enabled: true
+    schemaRegistry:
+      enabled: true # This tells the proto service to consider the schema registry when deserializing messages
+      refreshInterval: 5m # How often the compiled proto schemas in the cache should be updated
+----
+
+=== Local file system
+
+You can provide all required proto files by mounting them from your file system. All files must use the `.proto` file extension.
+In the `config.yaml` file, configure Redpanda Console to search one or more paths for your provided proto files, so it can
+build a registry with all types:
+
+[,yaml]
+----
+kafka:
+  protobuf:
+    enabled: true
+    mappings:
+      - topicName: orders
+        valueProtoType: fake_model.Order # You can specify the proto type for the record key and/or value (just one will work too)
+        keyProtoType: package.Type
+    fileSystem:
+      enabled: true
+      refreshInterval: 5m # 5min is the default refresh interval
+      paths:
+        - /etc/protos
+----
+
+=== GitHub repository
+
+To provide the proto files through a GitHub repository, first place the files in a directory. The directory doesn't matter, because
```

Comment:
You might want to  rephrase this -- "Redpanda will search the specified GitHub repository to a directory depth of up to five levels, so no specific directory name is required."